argo-jes/0.1.0/README.md

```diff
@@ -0,0 +1,279 @@
+      - list
+  # trafficsplit access needed for using the SMI provider
+  - apiGroups:
+      - split.smi-spec.io
+    resources:
+      - trafficsplits
+    verbs:
+      - create
+      - watch
+      - get
+      - update
+      - patch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: keptn-argo-service-rollouts
+  labels:
+    "app": "keptn"
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: keptn-argo-rollouts
+#  name: keptn-argo-service-rollouts
+subjects:
+  - kind: ServiceAccount
+    name: "jes-argo"
+    namespace: keptn-jes
+```
+
+Add the service account to your Kubernetes cluster:
+
+```bash
+kubectl apply -f serviceaccount.yaml
+```
+
+### Step 2: Add the Job-Executor configuration file
+
+The following configuration will allow you to run `argo rollouts promote` and `argo rollouts abort` using the Job-Executor-Service whenever the correct Keptn event is triggered.
+
+```yaml
+apiVersion: v2
+actions:
+  - name: "Run Argo promote"
+    events:
+      - name: "sh.keptn.event.release.triggered"
+        jsonpath:
+          property: "$.data.deployment.deploymentstrategy"
+          match: "user_managed"
+    tasks:
+      - name: "Execute Argo promote"
+        image: "quay.io/argoproj/kubectl-argo-rollouts"
+        args: ["promote", "$(SERVICE)-$(STAGE)", "-n", "$(PROJECT)-$(STAGE)"]
+        serviceAccount: "jes-argo"
+        env:
+          - name: SERVICE
+            value: "$.data.service"
+            valueFrom: event
+          - name: STAGE
+            value: "$.data.stage"
+            valueFrom: event
+          - name: PROJECT
+            value: "$.data.project"
+            valueFrom: event
+
+  - name: "Run Argo rollback"
+    events:
+      - name: "sh.keptn.event.rollback.triggered"
+    tasks:
+      - name: "Execute argo rollback"
+        image: "quay.io/argoproj/kubectl-argo-rollouts"
+        args: [ "abort", "$(SERVICE)-$(STAGE)", "-n", "$(PROJECT)-$(STAGE)" ]
+        serviceAccount: "jes-argo"
+        env:
+          - name: SERVICE
+            value: "$.data.service"
+            valueFrom: event
+          - name: STAGE
+            value: "$.data.stage"
+            valueFrom: event
+          - name: PROJECT
+            value: "$.data.project"
+            valueFrom: event
+
+  - name: "Argo test canary"
+    events:
+      - name: "sh.keptn.event.test.triggered"
+        jsonpath:
+          property: "$.data.test.teststrategy"
+          match: "real-user"
+    tasks:
+      - name: "Wait for canary wait duration"
+        image: "alpine"
+        cmd: [ "sleep" ]
+        args: [ "60s" ]
+```
+
+Add the job-executor configuration file to your Keptn service:
+
+```bash
+keptn add-resource --project=pod-tato-head --stage=production --service=helloservice --resource=job.config.yaml --resourceUri=job/config.yaml
+```
+
+Now the Job-Executor service will execute the Argo rollouts whenever you trigger a new evaluation.
```

Comment:
Please also mention that the subscriptions of the job-executor-service have to be adapted and maybe also list which ones are needed for the example above.